api/src/test/java/org/openmrs/api/ProgramWorkflowServiceTest.java

```diff
@@ -740,7 +750,262 @@ public void shouldTestPurgeProgramAttributeType() throws Exception {
+		cohort.getMemberIds().clear();
+		patientPrograms = dao.getPatientPrograms(null, programs);
+		assertEquals(patientPrograms.size(), 4);
+	}
+	
+	@Test
+	public void getPrograms_shouldTestGetProgramsIfCohortIsNotEmpty() {
+		int i = 0;
+		Cohort cohort = new Cohort();
+		Collection<Program> programs = pws.getAllPrograms();
+		List<PatientProgram> patientPrograms;
+		cohort.addMember(1);
+		cohort.addMember(2);
+		patientPrograms = dao.getPatientPrograms(cohort, programs);
+		assertEquals(patientPrograms.size(), 2);
+	}
+	
+	
+	@Test
+	public void voidPatientProgram_shouldTestVoidPatientProgram() {
+		PatientProgram patientProgram = pws.getPatientProgram(1);
+		PatientProgram patientProgram1 = pws.voidPatientProgram(patientProgram, "abc");
+		assertEquals(patientProgram1.getVoided(), true);
+		assertEquals(patientProgram1.getVoidReason(), "abc");
+	}
+	
+	@Test
+	public void unvoidPatientProgram_shouldTestUnvoidPatientProgram() {
+		//patientProgram retains a PatientProgram
+		PatientProgram patientProgram = pws.getPatientProgram(1);
+		
+		//patientProgramVoided is the patientProgram afeter voiding it
+		PatientProgram patientProgramVoided = pws.voidPatientProgram(patientProgram, "expired");
+		
+		//voidDate retains the voiding date for patientProgram
+		Date voidDate = patientProgram.getDateVoided();
+		patientProgramVoided = pws.unvoidPatientProgram(patientProgram);
+		assertEquals(patientProgramVoided.getVoided(), false);
+		for (PatientState state : patientProgram.getStates()) {
+			if (voidDate != null && voidDate.equals(state.getDateVoided())) {
+				assertEquals(state.getVoided(), false);
+				assertNull(state.getVoidedBy());
+				assertNull(state.getDateVoided());
+				assertNull(state.getVoidReason());
+			}
+		}
+	}
+	
+	@Rule
+	public ExpectedException expectedException1 = ExpectedException.none();
+	
+	@Test
+	public void saveConceptStateConversion_shouldThrowConceptStateConversionRequire() {
+		expectedException1.expect(APIException.class);
+		expectedException1.expectMessage("'ConceptStateConversion: Concept[null] results in State [null] for workflow [null]' failed to validate with reason: concept: Invalid concept, programWorkflow: Invalid Programme Workflow, programWorkflowState: Invalid Programme Workflow State");
+		ConceptStateConversion conceptStateConversion = new ConceptStateConversion();
+		conceptStateConversion.setConcept(null);
+		pws.saveConceptStateConversion(conceptStateConversion);
+	}
+	
+	@Test
+	public void saveConceptStateConversion_shouldTestSaveConceptStateConversion() {
+		//conceptStateConversion is the ConceptStateConversion we want to be saved
+		ConceptStateConversion conceptStateConversion = new ConceptStateConversion();
+		
+		/*conceptStateConversion1 is an existing ConceptStateConversion
+		we used this for taking some of it's attributes and assigning to 
+	'	the conceptStateConversion*/
+		ConceptStateConversion conceptStateConversion1 = pws.getAllConceptStateConversions().get(0);
```

Comment:
you can rename the variable as `existingConceptStateConversion`. No need to write explainations as comments. remove it please.

Reply:
Changed this